common/nym-lp/src/psk.rs

```diff
@@ -57,6 +114,349 @@ pub fn derive_psk(
+        .map_err(|e| LpError::Internal(format!("Ciphertext deserialization failed: {:?}", e)))?;
+
+    // Step 4: PSQ decapsulation for post-quantum security
+    let psq_psk = PsqX25519::decapsulate_psq(&kem_sk, &kem_pk, &ct, PSQ_SESSION_CONTEXT)
+        .map_err(|e| LpError::Internal(format!("PSQ decapsulation failed: {:?}", e)))?;
+
+    // Step 5: Combine ECDH + PSQ via Blake3 KDF (same formula as initiator)
+    let mut combined = Vec::with_capacity(64 + psq_psk.len());
+    combined.extend_from_slice(ecdh_secret.as_bytes());
+    combined.extend_from_slice(&psq_psk); // psq_psk is [u8; 32], need &
+    combined.extend_from_slice(salt);
+
+    let final_psk = nym_crypto::kdf::derive_key_blake3(PSK_PSQ_CONTEXT, &combined, &[]);
+
+    Ok(final_psk)
+}
+
+/// PSQ protocol wrapper for initiator (client) side.
+///
+/// Creates a PSQ initiator message with Ed25519 authentication, following the protocol:
+/// 1. Encapsulate PSK using responder's KEM key
+/// 2. Derive PSK and AEAD keys from K_pq
+/// 3. Sign the encapsulation with Ed25519
+/// 4. AEAD encrypt (timestamp || signature || public_key)
+///
+/// Returns (PSK, serialized_payload) where payload includes enc_pq and encrypted auth data.
+///
+/// # Arguments
+/// * `local_x25519_private` - Client's X25519 private key (for hybrid ECDH)
+/// * `remote_x25519_public` - Gateway's X25519 public key (for hybrid ECDH)
+/// * `remote_kem_public` - Gateway's PQ KEM public key (from KKT)
+/// * `client_ed25519_sk` - Client's Ed25519 signing key
+/// * `client_ed25519_pk` - Client's Ed25519 public key (credential)
+/// * `salt` - Session salt
+/// * `session_context` - Context bytes for PSQ (e.g., b"nym-lp-psq-session")
+///
+/// # Returns
+/// `(psk, psq_payload_bytes)` - PSK for Noise and serialized PSQ payload to embed
+pub fn psq_initiator_create_message(
+    local_x25519_private: &PrivateKey,
+    remote_x25519_public: &PublicKey,
+    remote_kem_public: &EncapsulationKey,
+    client_ed25519_sk: &ed25519::PrivateKey,
+    client_ed25519_pk: &ed25519::PublicKey,
+    salt: &[u8; 32],
+    session_context: &[u8],
+) -> Result<([u8; 32], Vec<u8>), LpError> {
+    // Step 1: Classical ECDH for baseline security
+    let ecdh_secret = local_x25519_private.diffie_hellman(remote_x25519_public);
+
+    // Step 2: PSQ v1 with Ed25519 authentication
+    // Extract X25519 KEM key from EncapsulationKey
+    let kem_pk = match remote_kem_public {
+        EncapsulationKey::X25519(pk) => pk,
+        _ => {
+            return Err(LpError::KKTError(
+                "Only X25519 KEM is currently supported for PSQ".to_string(),
+            ))
+        }
+    };
+
+    // Convert nym Ed25519 keys to libcrux format
+    type Ed25519VerificationKey = <Ed25519 as Authenticator>::VerificationKey;
+    let ed25519_sk_bytes = client_ed25519_sk.to_bytes();
+    let ed25519_pk_bytes = client_ed25519_pk.to_bytes();
+    let ed25519_verification_key = Ed25519VerificationKey::from_bytes(ed25519_pk_bytes);
```

Comment:
could we perhaps include feature-locked conversions with the `From` trait from nym types into libcrux?